### PR TITLE
fix(tooltip): Add delay to Tooltip show and hide

### PR DESCRIPTION
**Pull request by wooksauce**
## Summary

This PR introduces the default delay time of 300ms for showing a tooltip, and adds the ability to control the delay time for showing and hiding the tooltip.

category

Currently, Canvas-Kit tooltip doesn't support delayed showing and hiding. As a result, the tooltip appears as soon as the anchor item is hovered. I don't think that's the typical behavior of a tooltip, it's redundant most of the time, and makes the UI look busy. I chose 500ms as the default delay per my brief research (listed below in References) on tooltip behavior

### Release Note
This change could cause visual regression tests to fail if a screen shot is taken expecting a tooltip to show immediately. Your visual regression will either have to add an explicit wait of 300ms, or change the delay to 1ms only under test.

## Checklist

- [x] tests are changed or added
- [x] code has been documented

## Additional References

https://www.nngroup.com/articles/timing-exposing-content/
https://www.linkedin.com/pulse/tooltips-samson-vijay-kumar-tennela/
https://ux.stackexchange.com/questions/358/how-long-should-the-delay-be-before-a-tooltip-pops-up

### CODE REVIEW

File: modules/react/tooltip/spec/Tooltip.spec.tsx [hunk 1]
```diff
@@ -59,6 +62,25 @@ describe('Tooltip', () => {
...
+      fireEvent.mouseEnter(screen.getByText('Test Text')); // triggers the tooltip
+
+      expect(screen.queryByText('Delayed Tooltip Text')).toBeNull(); // tooltip is not shown before the delay
+
+      jest.advanceTimersByTime(1000); // advance the timer by the amount of delay time
+      expect(screen.getByText('Delayed Tooltip Text')).toBeInTheDocument();
+    });
+    jest.clearAllTimers();
+  });
+
```
Comment (csongnguyen) [inline modules/react/tooltip/spec/Tooltip.spec.tsx:83]: we should include a test for hideDelay as well

File: modules/react/tooltip/lib/Tooltip.tsx [hunk 2]
```diff
@@ -66,10 +74,17 @@ export const Tooltip = ({
   placement = 'top',
   title,
   children,
+  showDelay = 500,
```
Comment (NicholasBoll) [inline modules/react/tooltip/lib/Tooltip.tsx:77]:
```suggestion
  showDelay = 300,
```

File: modules/react/tooltip/lib/useTooltip.tsx [hunk 3]
```diff
@@ -54,6 +54,8 @@ const isInteractiveElement = (element: Element) => {
 export function useTooltip<T extends Element = Element>({
   type = 'label',
   titleText = '',
+  showDelay = 500,
```
Comment (NicholasBoll) [inline modules/react/tooltip/lib/useTooltip.tsx:57]:
```suggestion
  showDelay = 300,
```

File: modules/react/tooltip/spec/Tooltip.spec.tsx [hunk 4]
```diff
@@ -26,11 +27,13 @@ describe('Tooltip', () => {
 
       fireEvent.mouseEnter(screen.getByText('Test Text')); // triggers the tooltip
 
+      jest.advanceTimersByTime(500); // advance the timer by the amount of delay time
```
Comment (NicholasBoll) [inline modules/react/tooltip/spec/Tooltip.spec.tsx:30]:
```suggestion
      jest.advanceTimersByTime(300); // advance the timer by the amount of delay time
```

File: modules/react/tooltip/spec/useTooltip.spec.tsx [hunk 5]
```diff
@@ -25,16 +25,19 @@ describe('useTooltip with type="label"', () => {
...
 describe('useTooltip with type="describe"', () => {
+  jest.useFakeTimers();
   it('should add aria attributes to correlate the target and the tooltip', () => {
     const {getByText, getByRole} = render(<TooltipWithHook type="describe" />);
 
     const target = getByText('Hover');
     const tooltip = getByRole('tooltip');
 
     fireEvent.mouseOver(target); // assign the ID to the tooltip
+    jest.advanceTimersByTime(500); // advance the timer by the amount of delay time
```
Comment (NicholasBoll) [inline modules/react/tooltip/spec/useTooltip.spec.tsx:36]:
```suggestion
    jest.advanceTimersByTime(300); // advance the timer by the amount of delay time
```

File: modules/react/tooltip/stories/Tooltip.stories.mdx [hunk 6]
```diff
@@ -90,6 +91,13 @@ a dialog instead.
 
 <ExampleCodeBlock code={CustomContent} />
 
+### Delayed Tooltip
+
+The default delay for showing and hiding a tooltip are 500ms and 100ms, respectively. You can
```
Comment (NicholasBoll) [inline modules/react/tooltip/stories/Tooltip.stories.mdx:96]:
```suggestion
The default delay for showing and hiding a tooltip are 300ms and 100ms, respectively. You can
```

File: modules/react/tooltip/spec/Tooltip.spec.tsx [hunk 7]
```diff
@@ -59,6 +62,25 @@ describe('Tooltip', () => {
...
+        <Tooltip type="describe" title="Delayed Tooltip Text" showDelay={1000}>
+          <span>Test Text</span>
+        </Tooltip>
+      );
+
+      fireEvent.mouseEnter(screen.getByText('Test Text')); // triggers the tooltip
+
+      expect(screen.queryByText('Delayed Tooltip Text')).toBeNull(); // tooltip is not shown before the delay
+
+      jest.advanceTimersByTime(1000); // advance the timer by the amount of delay time
```
Comment (NicholasBoll) [inline modules/react/tooltip/spec/Tooltip.spec.tsx:78]: This is validating the delay is at least 1000ms and more than 0, but not that the override works correctly. The following will fix that:

```suggestion

      jest.advanceTimersByTime(300); // advance the timer by the default delay time
      expect(screen.queryByText('Delayed Tooltip Text')).toBeNull(); // tooltip is not shown before the configured delay

      jest.advanceTimersByTime(700); // advance the timer by the amount of delay time
```

**Review by NicholasBoll — APPROVED** [review]

**Pull request merged**